packages/opentelemetry-core/src/utils/environment.ts

```diff
@@ -58,15 +130,15 @@ export const DEFAULT_ENVIRONMENT: Required<ENVIRONMENT> = {
  * @param max
  */
 function parseNumber(
-  name: keyof ENVIRONMENT,
-  environment: ENVIRONMENT_MAP | ENVIRONMENT,
-  values: ENVIRONMENT_MAP,
+  name: keyof ENVIRONMENT_NUMBERS,
+  environment: ENVIRONMENT,
+  values: RAW_ENVIRONMENT,
   min = -Infinity,
   max = Infinity
 ) {
-  if (typeof values[name] !== 'undefined') {
-    const value = Number(values[name] as string);
-
+  const givenValue = values[name];
+  if (typeof givenValue === 'string' || typeof givenValue === 'number') {
```

Comment:
Apparently I was mistaken about typescript issues. I looked at a couple approaches to typing and probably left it because it was needed in a different scenario. It's not necessary know, so reverting.